content/automate/ManageIQ/Infrastructure/VM/Transform/Import.class/__methods__/create_vm_import_request.rb

```diff
@@ -13,30 +13,67 @@ def initialize(handle = $evm)
                 @handle = handle
               end
 
+              def initialize_storage_map(provider_id)
+                @storage_map = {}
+                provider = @handle.vmdb(:ext_management_system, provider_id)
+                return unless provider.present?
+                provider.storages.each do |storage|
+                  tag = storage_mapping_tag(storage)
+                  next unless tag.present?
+                  raise_error("Error, several storages are tagged with tag #{tag}") if @storage_map.key?(tag)
+                  @storage_map[tag] = storage.id
+                  @handle.log(:info, "Storage with tag #{tag} is mapped to target storage #{storage.name}")
+                end
+              end
+
               def main
-                validate_root_args(%w(vm dialog_provider dialog_cluster dialog_storage dialog_sparse))
+                validate_root_args(%w(dialog_provider dialog_cluster dialog_sparse))
                 if @handle.root['dialog_tag_category'].present? && @handle.root['dialog_tag_name'].present?
                   tag = "/#{@handle.root['dialog_tag_category']}/#{@handle.root['dialog_tag_name']}"
                   tagged_vms = @handle.vmdb('ManageIQ_Providers_Vmware_InfraManager_Vm')
                                       .find_tagged_with(:all => tag, :ns => '/managed')
+                  initialize_storage_map(@handle.root['dialog_provider'])
                   tagged_vms.each do |vm|
-                    create_request(vm, '')
+                    target_storage_id = nil
+                    if vm.storage.present?
+                      tag = storage_mapping_tag(vm.storage)
+                      raise_error("Error, cannot determine the target storage for storage #{vm.storage.name} without tag") unless tag.present?
+                      target_storage_id = @storage_map[tag]
+                      raise_error("Error, no target storage with tag #{tag}") unless target_storage_id.present?
+                    end
+                    create_request(vm, '', target_storage_id)
```

Comment:
This implies the `dialog_storage` will be ignored if VMs are selected through tagging. Is this what the user expects - unless the dialog will hide the selection of storage?